### ISSUE TITLE
Deleting surveys makes it impossible to create a new survey for that event

### ISSUE THREAD

**Issue by edvardvb** [webkom/lego#1148]
Interkom will probably delete a survey and create a new one at some point, instead of editing the existing survey. This leads to 409 upon trying to create a new survey, because they are persistent, and the event field is one-to-one.

**Comment by odinuge** [comment]
We can make the field `event` nullable (`null=Trure`), and set null on delete. Or, any thoughts?

**Comment by finnss** [comment]
Sure, that works. You'll just have to make sure you handle someone trying to access a survey with `event=null`, because that would likely crash both the API and the Webapp right now. I've assumed each survey has an event everywhere. 

Shouldn't be worse than checking for it early in the backend call, and immedeatly returning an error message (404?) if `event=null`. Then of course displaying the error message frontend, which I guess is free if we use 404.

And of course, any endpoint returning a list of surveys would have to filter out any with `event=null`. Maybe Django does that for you when you set `deleted=True`, I don't know.

**Comment by odinuge** [comment]
Yeah, there will be quite a few edge cases we would need to think about :upside_down_face: We should probably make a list or something, before we start with it.

When we run `survey.delete()` the `PeristantModel` would just set `deleted=True` like you say, and it would be filtered out by default. The elements can still be reached by using `Survey.all_objects` but no views use those. 

Or is it better to just use `.delete(force=true)` to properly delete it on sql level?? :thinking:

**Comment by finnss** [comment]
Yeah of course we talked about that as well, essentially having it not be persistent, would certainly solve this issue. But the argument for keeping it persistent was that accidentally deleting a survey would also delete every answer (assuming cascading delete), which is exactly the kind of data we want to save with persistency, right?

You could always have the survey itself be non-persistent but the answers persistent, but then you run into rather wierd situations where you might not know what question an answer belongs to, or if two questions belonged to the same survey, etc. The answers lose their value without the survey, I think. So we decided to keep it all persistent. 

I think making event nullable and handling the edge cases is a better approach :) 

As for edge cases to think about, should a deleted event "remember" which event it belonged to?

**Comment by edvardvb** [comment]
Hacky workaround - have a separate button/endpoint for creating new surveys to overwrite the old one 🙃
